### ISSUE TITLE
GamepadEventsRaw vs GamepadEvents is confusing and undocumented

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#5544]
## What problem does this solve or what need does it fill?

When mocking gamepad input, you *must* send `GamepadEventsRaw`, or it will not be transformed into `Input` and `Axis` data correctly.

However, there are not docs to explain the difference or warn users of this footgun.

## What solution would you like?

1. Add documentation to `GamepadEvents` and `GamepadEventsRaw` explaining their purpose.
2. Add a doc test demonstrating how to use `GamepadEventsRaw` to mock gamepad input.

**Comment by alice-i-cecile** [comment]
I'd like to reconsider this design down the road, or at least give these more descriptive names, but docs should be done first to make that work easier to review.

**Comment by KDecay** [comment]
Going to document the `gamepad.rs` file since I have already done this in  #3692 but didn't make a new PR for it yet.
